### PR TITLE
Revise Git commit hash handling

### PR DESCRIPTION
**Pull request by rubo**
## Changes

- Removed dependency on Git executable in project files
- Added `Microsoft.Build.Tasks.Git` package to incorporate the commit hash into the version string
- Stopped copying the `.git` directory to Docker containers, which significantly increases local Docker image build times
- Revised `ClientVersionV1` tests
- Removed the branch name check from the project file so as not to fail the build because of an undesired branch name. This logic is completely unrelated to the build process and must be handled at the GitHub level

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [x] Refactoring
- [ ] Documentation update
- [x] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Requires documentation update

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Core/ProductInfo.cs [hunk 1]
```diff
@@ -4,44 +4,49 @@
...
+        var index = version.IndexOf('+', StringComparison.Ordinal);
+
+        if (index == -1)
+            Version = version;
+        else
+        {
+            Commit = version[(index + 1)..];
+            Version = version[..Math.Min(index + 9, version.Length - 1)];
+        }
+
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/ProductInfo.cs:39]:
```suggestion
        Version = versionAttr.InformationalVersion;
        var index = Version.IndexOf('+', StringComparison.Ordinal);

        if (index != -1)
        {
            Commit = Version[(index + 1)..];
            Version = Version[..Math.Min(index + 9, version.Length - 1)];
        }

```

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Runner/Nethermind.Runner.csproj [hunk 2]
```diff
@@ -106,25 +110,4 @@
...
-        <_Parameter2>$(Commit)</_Parameter2>
-      </AssemblyAttribute>
-    </ItemGroup>
-  </Target>
-
-  <Target Name="CheckBranchName" AfterTargets="BeforeBuild">
-    <Exec Command="git branch --show-current" ConsoleToMSBuild="True" StandardOutputImportance="low">
-      <Output PropertyName="BranchName" TaskParameter="ConsoleOutput" />
-    </Exec>
-    <Error Text="The current branch '$(BranchName)' should be named in lower case." Condition="$([System.Text.RegularExpressions.Regex]::IsMatch($(BranchName), '^.*[A-Z].*$'))" />
```
Comment (flcl42) [inline src/Nethermind/Nethermind.Runner/Nethermind.Runner.csproj:128]: Will it be checked somewhere else?

Comment (rubo) [reply src/Nethermind/Nethermind.Runner/Nethermind.Runner.csproj:128]: Yes, with a GitHub ruleset, so it won't be optional as it is now and won't mess with the build process.

**Review by smartprogrammer93 — APPROVED** [review]

**Comment by michaelsproul** [comment]
Was Nethermind returning a commit hash with length < 8 prior to this change? We are getting errors connecting LH to Nethermind v1.26:

> Jun 06 03:04:48.001 WARN Execution engine call failed            error: InvalidClientVersion("Input must be exactly 8 characters long (excluding any '0x' prefix)"), service: exec

**Comment by smartprogrammer93** [comment]
@michaelsproul it used to return longer than 8:
https://github.com/NethermindEth/nethermind/pull/7013

**Pull request merged**